src/libraries/Directory.Build.targets

```
@@ -42,6 +42,13 @@
     <IsNETCoreAppRef Condition="'$(IsNETCoreAppRef)' == '' and
                                 '$(IsReferenceAssembly)' == 'true' and
                                 $(NetCoreAppLibrary.Contains('$(AssemblyName);'))">true</IsNETCoreAppRef>
+
+    <AnalyzerForAssembly Condition="'$(AnalyzerForAssembly)' == '' AND ('$(IsGeneratorProject)' == 'true' or '$(IsAnalyzerProject)' == 'true')">$(AssemblyName.Substring(0, $(AssemblyName.LastIndexOf('.'))))</AnalyzerForAssembly>
```

nit: we usually use lower case operators:

```suggestion
    <AnalyzerForAssembly Condition="'$(AnalyzerForAssembly)' == '' and ('$(IsGeneratorProject)' == 'true' or '$(IsAnalyzerProject)' == 'true')">$(AssemblyName.Substring(0, $(AssemblyName.LastIndexOf('.'))))</AnalyzerForAssembly>
```

---

I was wondering if you couldn't use the GetFilePathWithoutExtension function here to strip the "extension" (which in this case of course isn't a file extension) away?

---

Sure I can do that. What’s your opinion about this vs some other convention? We could find the directory under libraries and use that for lookup.  We could have a direct property for analyzers and avoid the lookup.
I’m thinking the latter might be simpler.

---

Whatever you prefer. I'm fine with either the convention or with the inlined property.

---

I think I prefer the simpler convention of `IsNETCoreAppAnalyzer` and `IsAspNetCoreAppAnalyzer` in the projects themselves.  Its less magic.